### PR TITLE
Update Gradle plugins build

### PR DESCRIPTION
**Pull request by grossws**
* migration to Gradle Kotlin DSL (much more streamlined IDE support)
* use convention plugins in `build-logic/src/main/kotlin` to simplify actual `build.gradle.kts` build scripts
* use version catalog for deps (see https://docs.gradle.org/current/userguide/platforms.html#header)
* ~use java toolchain to build plugins (see https://docs.gradle.org/current/userguide/toolchains.html)~ **UPD** use `JavaCompile#options#release` to set javac `--release` option to effectively consistently configure source/target compatibility and bootstrap classpath
* use `JavaPluginExtension#withSourcesJar`/`withJavadocJar` instead of older packaging them manually
* update gradle wrapper (**UPD**: for both `devtools/gradle` and `integration-tests/gradle`)
* fix dependencies on gradle plugin markers in `integration-tests/gradle`, `<type>pom</type>` should be used since `com.gradle.plugin-publish` plugin publishes only `pom.xml` to `io.quarkus:io.quarkus.gradle.plugin` GA coords (see https://docs.gradle.org/current/userguide/plugins.html#sec:plugin_markers)
* add `org.bouncycastle:bcpkix-jdk18on` to integration test fixtures in `integration-tests/gradle` to fix test failure when ECDSA keys/certs present in local `.kube/config`
* use `task.getLogger()` instead of `task.getProject().getLogger()` (little bit towards potential configuration cache compatibility)

### CODE REVIEW

**Comment by gsmet** [comment]
@snazy could you have a look at this one?

**Comment by grossws** [comment]
At least I should use `options.release` instead of toolchains. 

And since Quarkus 3.x requires java 11+ we could drop `sourceCompatibility`/`targetCompatibility` and use `release` instead.

**Comment by snazy** [comment]
Hm, I guess, the toolchain-change is required (see CI):
```
Caused by: org.gradle.jvm.toolchain.internal.ToolchainDownloadFailedException: No locally installed toolchains match (see https://docs.gradle.org/8.1/userguide/toolchains.html#sec:auto_detection) and toolchain download repositories have not been configured (see https://docs.gradle.org/8.1/userguide/toolchains.html#sub:download_repositories).
```

**Comment by grossws** [comment]
Yeah, it seems gradle discovered where tooling cache lives on Linux runners but not on Windows ones.

**Comment by grossws** [comment]
PR updated:

* wrapper in `integration-tests/gradle` with dummy `build.gradle.kts` and `settings.gradle.kts` to simplify future updates 
* reverted to setting `--release` in `JavaCompile` tasks instead of using newer java toolchains (since Quarkus requires Java 11 to build now `sourceCompatibility`/`targetCompatibility` are unnecessary when `options.release` is set)

File: integration-tests/gradle/gradle/wrapper/gradle-wrapper.properties [hunk 1]
```diff
@@ -1,6 +1,5 @@
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-# https://gradle.org/release-checksums/
```
Comment (snazy) [inline integration-tests/gradle/gradle/wrapper/gradle-wrapper.properties:3]: Can you leave this line? It's pretty handy to know where to cross-check the checksum and where to get the ones during updates.

File: integration-tests/gradle/build.gradle.kts [hunk 2]
```diff
@@ -0,0 +1,5 @@
+tasks.wrapper {
```
Comment (snazy) [inline integration-tests/gradle/build.gradle.kts:1]: Can you add a comment that this (and `settings.gradle.kts` are rather "dummy" files?

**Review by aloubyansky — APPROVED** [review]
Thanks @grossws

**Review by snazy — COMMENTED** [review]
Really minor nits - current state of the PR looks good to me. +1

File: integration-tests/gradle/gradle/wrapper/gradle-wrapper.properties (hunk 1 above)
Comment (grossws) [reply integration-tests/gradle/gradle/wrapper/gradle-wrapper.properties:3]: Sorry, missed it. Gradle's `Wrapper` task rewrites properties and I missed this change

File: integration-tests/gradle/build.gradle.kts (hunk 2 above)
Comment (grossws) [reply integration-tests/gradle/build.gradle.kts:1]: since it's already merged should I create separate PR?

Comment (aloubyansky) [reply integration-tests/gradle/build.gradle.kts:1]: Yes, please @grossws. Thanks.

Comment (grossws) [reply integration-tests/gradle/build.gradle.kts:1]: Ok, will create new PR soon

**Pull request merged**